fuzzers/patch_parse_fuzzer.c

```diff
@@ -0,0 +1,39 @@
+/*
+ * libgit2 patch parser fuzzer target.
+ *
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+
+#include "git2.h"
+#include "patch.h"
+#include "patch_parse.h"
+
+#define UNUSED(x) (void)(x)
+
+int LLVMFuzzerInitialize(int *argc, char ***argv)
+{
+	UNUSED(argc);
+	UNUSED(argv);
+
+	if (git_libgit2_init() < 0)
+		abort();
+
+	return 0;
+}
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
+{
+	if (size) {
+		git_patch *patch;
+		git_patch_options opts = {(uint32_t)data[0]};
+		git_patch_from_buffer(&patch, (const char *)data + 1, size - 1,
+		                      &opts);
```

Comment:
If this function returns an error, libgit2 doesn't guarantee that `patch` has been initialized to `NULL`. Thus you either need to check for the return value of this function or otherwise initialize `patch = NULL`